### PR TITLE
show 頁面 搜尋框 改善

### PR DESCRIPTION
**Pull request by homerchen19**
to fix #310

### CODE REVIEW

**Comment by mark86092** [comment]
Please don't merge `goodjoblife:dev` into your branch `improve-search-bar` (it has no conflict), can you reset it?

File: src/js/main/show.js [hunk 1]
```diff
@@ -153,20 +153,52 @@ const app = new Vue({
...
+        router.setRoute(`/search-and-group/by-job-title/${encodeURIComponent(this.keyword)}`);
+      } else {
+        router.setRoute("/latest");
+      }
+    },
+    setDefault: function() {
+      this.search_type = "by-company";
+      this.keyword = "";
+    },
+    setInputInfo: function(search_type, keyword) {
```
Comment (mark86092) [inline src/js/main/show.js]: you can use `function(search_type = "by-company", keyword = "")`, we use babel to compile it

File: src/js/main/show.js [hunk 2]
```diff
@@ -153,20 +153,52 @@ const app = new Vue({
...
+});
+
 const router = Router({
   "/latest": function() {
     app.currentView = "latestWorkings";
+    searchBarApp.setDefault();
   },
   "/search-and-group/by-job-title/(.*)": function(name) {
     app.currentView = "searchAndGroupByJobTitle";
+    let deCodeName = decodeURIComponent(name);
```
Comment (mark86092) [inline src/js/main/show.js]: `const decodeName` or `const decode_name`

File: src/js/main/show.js (hunk 1 above)
Comment (homerchen19) [reply src/js/main/show.js]: I don't get it.
It will have error when I change to `function(search_type = "by-company", keyword = "")`

Comment (mark86092) [reply src/js/main/show.js]: what's the error?

**Comment by homerchen19** [comment]
Did I really reset it ?
Sorry, I'm not sure whether what I did is correct QQ

File: src/js/main/show.js (hunk 1 above)
Comment (homerchen19) [reply src/js/main/show.js]: Not be able to compile

Comment (mark86092) [reply src/js/main/show.js]: I use this, and get 

image

image

Comment (mark86092) [reply src/js/main/show.js]: 這樣寫的目的是可以把 setDefault 取代成一般的情況

Comment (homerchen19) [reply src/js/main/show.js]: Oh yes. Sorry I misunderstand what your mean haha.
This is great

**Comment by mark86092** [comment]
git 的指令我再跟你用 slack 說明

**Review by mark86092 — APPROVED** [review]
看起來ＯＫ的，再給 @barry800414 review 一下

File: src/js/main/show.js (hunk 2 above)
Comment (barry800414) [reply src/js/main/show.js]: I suggest `decoded_name` or `decodedName`

**Review by barry800414 — CHANGES_REQUESTED** [review]
除了變數名稱要修一下，我想沒有問題! 
另外想問的是，搜尋空白字元這件事，有要在這個PR做嗎?

**Comment by mark86092** [comment]
@barry800414 關於空白，我想先在 https://github.com/goodjoblife/WorkTimeSurvey/blob/dev/src/views/show.pug#L29

`input(required)` 這裡做處理好了（新 PR）

**Comment by homerchen19** [comment]
所以我改一下變數名稱就OK了嗎～
還是連input(required)也要

**Comment by mark86092** [comment]
input 那個另外處理

**Comment by mark86092** [comment]
@barry800414 The PR is updated! tks

File: src/js/main/show.js [hunk 3]
```diff
@@ -169,15 +169,15 @@ const router = Router({
   },
   "/search-and-group/by-job-title/(.*)": function(name) {
     app.currentView = "searchAndGroupByJobTitle";
-    const deCodeName = decodeURIComponent(name);
+    const decodedName = decodeURIComponent(name);
     searchBarApp.setInputInfo("by-job-title", deCodeName);
```
Comment (barry800414) [inline src/js/main/show.js:173]: 後面的變數名稱沒改，連搜尋都不能搜尋。
之後有改一定要先測試過。

**Pull request merged**